Lib/socket.py

```diff
@@ -728,6 +728,78 @@ def create_connection(address, timeout=_GLOBAL_DEFAULT_TIMEOUT,
     else:
         raise error("getaddrinfo returns an empty list")
 
+def bind_socket(address, family=AF_UNSPEC, type=SOCK_STREAM, *, backlog=128,
+                reuse_port=False, flags=getattr(_socket, "AI_PASSIVE", 0)):
+    """Convenience function which creates a socket bound to *address*
+    (a 2-tuple (host, port)) and return the socket object.
+
+    If *host* is an empty string or None all network interfaces are assumed.
+
+    If *family* is AF_UNSPEC or None the address family will be
+    determined from the *host* specified in *address*.
+
+    *type* should be either SOCK_STREAM or SOCK_DGRAM.
+
+    *backlog* is the queue size passed to socket.listen() and is ignored
+    for SOCK_DGRAM socket types.
+
+    *flags* is a bitmask for getaddrinfo().
+
+    >>> with bind_socket((None, 8000)) as server:
+    ...     while True:
+    ...         conn, addr = server.accept()
+    ...         # handle new connection
+    """
+    host, port = address
+    if host == "":
+        # https://mail.python.org/pipermail/python-ideas/2013-March/019937.html
+        host = None  # all interfaces
+    if not family:
```

Comment:
Technically `[]` and `0.0` are acceptable.
I don't insist on argument type check in runtime here, avoiding `None` would be enough.